### PR TITLE
[1/6] Deployments Table - Deployment struct updates

### PR DESCRIPTION
**Pull request by philipnrmn**
Note that this work is based off a feature branch (`deployments-table`).

This PR adds the `affectedServices` field to `Deployment`. This field is populated by `DCOSStore` when it receives deployments data from `MarathonStore`. If it is not populated and something attempts to access it, it will error rather than show bad data. 

Once this is merged, the whole services table can be populated using only the output from `DCOSStore.deploymentsList`.

### CODE REVIEW

File: src/js/stores/DCOSStore.js [hunk 1]
```diff
@@ -79,6 +89,25 @@ class DCOSStore extends EventEmitter {
...
+  onMarathonDeploymentsChange() {
+    let deploymentsList = MarathonStore.get('deployments');
+    let serviceTree = MarathonStore.get('groups');
+
+    // Populate deployments wth affected services
+    this.data.marathon.deploymentsList = deploymentsList
+      .mapItems(function (deployment) {
+        let ids = deployment.getAffectedServiceIds();
+        let services = ids.map(function (id) {
+          console.log(`Deployment: ${deployment.id} ; Service: ${id}`);
```
Comment (kennyt) [inline src/js/stores/DCOSStore.js]: console.log here. did you want to leave that in?

File: src/js/stores/DCOSStore.js [hunk 2]
```diff
@@ -79,6 +89,25 @@ class DCOSStore extends EventEmitter {
     MarathonStore.fetchServiceVersions(serviceID);
   }
 
+  onMarathonDeploymentsChange() {
+    let deploymentsList = MarathonStore.get('deployments');
+    let serviceTree = MarathonStore.get('groups');
+
+    // Populate deployments wth affected services
+    this.data.marathon.deploymentsList = deploymentsList
```
Comment (kennyt) [inline src/js/stores/DCOSStore.js]: we can just do `this.deploymentsList` because we have a getter for this, right?

File: src/js/stores/DCOSStore.js (hunk 1 above)
Comment (philipnrmn) [inline src/js/stores/DCOSStore.js]: I did not. Thanks!

**Comment by mesosphere-web** [comment]
Reviewer: @MatApple

File: src/js/stores/DCOSStore.js [hunk 3]
```diff
@@ -3,18 +3,22 @@ import {EventEmitter} from 'events';
...
   MESOS_SUMMARY_CHANGE,
+  MARATHON_DEPLOYMENTS_CHANGE,
   MARATHON_GROUPS_CHANGE,
   MARATHON_SERVICE_VERSION_CHANGE,
   MARATHON_SERVICE_VERSIONS_CHANGE
 } from '../constants/EventTypes';
 import Framework from '../structs/Framework';
 import MarathonStore from './MarathonStore';
 import MesosSummaryStore from './MesosSummaryStore';
+import DeploymentsList from '../structs/DeploymentsList';
```
Comment (rcorral) [inline src/js/stores/DCOSStore.js]: 🔤 !

File: src/js/stores/DCOSStore.js [hunk 4]
```diff
@@ -79,6 +89,24 @@ class DCOSStore extends EventEmitter {
     MarathonStore.fetchServiceVersions(serviceID);
   }
 
+  onMarathonDeploymentsChange() {
+    let deploymentsList = MarathonStore.get('deployments');
+    let serviceTree = MarathonStore.get('groups');
+
+    // Populate deployments wth affected services
```
Comment (MatApple) [inline src/js/stores/DCOSStore.js]: #nit typo `wth` -> `with` :)

File: src/js/stores/DCOSStore.js [hunk 5]
```diff
@@ -79,6 +89,24 @@ class DCOSStore extends EventEmitter {
...
+    this.data.marathon.deploymentsList = deploymentsList
+      .mapItems(function (deployment) {
+        let ids = deployment.getAffectedServiceIds();
+        let services = ids.map(function (id) {
```
Comment (MatApple) [inline src/js/stores/DCOSStore.js]: Could shorten this to just `ids.map(serviceTree.findItemById);` as an alternative :)

**Comment by MatApple** [comment]
Nice @philipnrmn!

**Comment by philipnrmn** [comment]
Thanks for the speedy review! Addressed all comments.

**Comment by philipnrmn** [comment]
Incidentally, integration tests are failing because something's up with Velocity - they are certified to be Working On My Machine™

**Comment by pierlo-upitup** [comment]
🚢  great stuff!
